apps/labs/posts/hypothesis-array-api.md

```diff
@@ -0,0 +1,713 @@
+</pre>
+```
+
+```{=html}
+<pre style="white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace"><span style="color: #808000; text-decoration-color: #808000">...</span>
+</pre>
+```
+
+How Hypothesis \"draws\" from its strategies can look rather
+unremarkable at first. A small sample of draws might look fairly uniform
+but trust that strategies will end up covering all kinds of edge cases.
+Importantly it will cover these cases efficiently so that
+Hypothesis-powered tests are *relatively* quick to run on your machine.
+
+All our test method does is get the cumulative sums array `a` that is
+returned from `cumulative_sums(x)`, and then check that every element
+`a[i]` is greater than or equal to `a[i-1]`.
+
+Time to run it!
+
+``` python
+ipytest.run("-k positive_arrays_have_incrementing_sums", "--hypothesis-seed=3")
+```
+
+    F                                                                        [100%]
+    =================================== FAILURES ===================================
+    _________________ test_positive_arrays_have_incrementing_sums __________________
+
+        @given(xps.arrays(dtype="uint8", shape=10))
+    >   def test_positive_arrays_have_incrementing_sums(x):
+
+    <cell>:7: 
+    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
+
+    x = Array([26, 26, 26, 26, 26, 26, 26, 26, 26, 26], dtype=uint8)
+
+        @given(xps.arrays(dtype="uint8", shape=10))
+        def test_positive_arrays_have_incrementing_sums(x):
+            a = cumulative_sums(x)
+    >       assert nxp.all(a[1:] >= a[:-1])
+    E       assert Array(False, dtype=bool)
+    E        +  where Array(False, dtype=bool) = <function all at 0x7f2d48cc2430>(Array([ 52,  78, 104, 130, 156, 182, 208, 234,   4], dtype=uint8) >= Array([ 26,  52,  78, 104, 130, 156, 182, 208, 234], dtype=uint8))
```

Comment:
I'm getting the Vercel build error: `error [Error: [next-mdx-remote] error compiling MDX:`

Full error message:
```
Unexpected character `.` (U+002E) in attribute name, 
expected an attribute name character such as letters, 
digits, `$`, or `_`; `=` to initialize a value; whitespace before attributes; or the end of the tag

  74 |
  75 |     ERROR: not found: /home/user/tmp35zsok9u.py::TestFoo::test_bar
> 76 |     (no name '/home/user/tmp35zsok9u.py::TestFoo::test_bar' in any of [<Module tmp35zsok9u.py>])
     |                                                                                           ^
  77 |
  78 | If the absence of this very basic [`pytest`](https://pytest.org) feature
```

[Link to error](https://vercel.com/quansight/labs/D8SyDYayvih2edfQoRigFpsgdrWZ)

The context is showing the result of some code. [Link to this section in the old blog website.](https://labs.quansight.org/blog/2021/06/pytest-pytorch#:~:text=collected%20in%200.01s-,ERROR,-%3A%20not%20found%3A%20/home)

Similar problem to #647: output of code blocks formatted as plain text, not in code blocks, causing problems to Vercel